Stop exposing showModalDialog(), behind a linked-on-after check

cdumez opened the pull request:
#### c4bb4d1af89e1ef5f57ef1b1cf7b65ce62c30ba5
<pre>
Stop exposing showModalDialog(), behind a linked-on-after check
<a href="https://bugs.webkit.org/show_bug.cgi?id=243099">https://bugs.webkit.org/show_bug.cgi?id=243099</a>
&lt;rdar://97441232&gt;

Reviewed by Ryosuke Niwa and Darin Adler.

Now that we support the dialog element, drop support for showModalDialog(),
behind a linked-on-after check. Blink and Gecko have already dropped their
support for showModalDialog() a long time ago.

* LayoutTests/editing/execCommand/show-modal-dialog-during-execCommand.html:
* LayoutTests/fast/animation/request-animation-frame-during-modal.html:
* LayoutTests/fast/dom/Window/delete-operations.html:
* LayoutTests/fast/dom/Window/forbid-showModalDialog.html:
* LayoutTests/fast/dom/Window/open-window-min-size.html:
* LayoutTests/fast/dom/Window/showModalDialog-mandatory-parameter.html:
* LayoutTests/fast/dom/Window/window-function-frame-getter-precedence.html:
* LayoutTests/fast/dom/Window/window-function-name-getter-precedence.html:
* LayoutTests/fast/dom/Window/window-lookup-precedence.html:
* LayoutTests/fast/dom/Window/window-property-shadowing.html:
* LayoutTests/fast/dom/null-page-show-modal-dialog-crash.html:
* LayoutTests/fast/dom/wrapper-identity.html:
* LayoutTests/fast/events/beforeunload-showModalDialog.html:
* LayoutTests/fast/events/pagehide-showModalDialog.html:
* LayoutTests/fast/events/scroll-event-during-modal-dialog.html:
* LayoutTests/fast/events/show-modal-dialog-onblur-onfocus.html:
* LayoutTests/fast/events/unload-showModalDialog.html:
* LayoutTests/fast/harness/show-modal-dialog.html:
* LayoutTests/http/tests/cookies/document-cookie-after-showModalDialog.html:
* LayoutTests/http/tests/security/cross-frame-access-call.html:
* LayoutTests/http/tests/security/cross-frame-access-get.html:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor.html:
* LayoutTests/http/tests/security/cross-frame-access-put.html:
* LayoutTests/http/tests/security/cross-origin-modal-dialog-base.html:
* LayoutTests/http/tests/security/navigate-when-restoring-cached-page.html:
* LayoutTests/http/tests/security/showModalDialog-sync-cross-origin-page-load.html:
* LayoutTests/http/tests/security/showModalDialog-sync-cross-origin-page-load2.html:
* LayoutTests/js/dom/function-length.html:
* Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
(WebCore::JSDOMWindowBase::finishCreation):
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultShowModalDialogEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:
(WebKit::defaultShowModalDialogEnabled):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
(-[ModalDelegate _webViewRunModal:]): Deleted.
(-[ModalDelegate _webView:setResizable:]): Deleted.
(-[ModalDelegate webView:createWebViewWithConfiguration:forNavigationAction:windowFeatures:]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestUIClient.cpp:
(beforeAll):
(testWebViewAllowModalDialogs): Deleted.
(testWebViewDisallowModalDialogs): Deleted.
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):

Canonical link: <a href="https://commits.webkit.org/252759@main">https://commits.webkit.org/252759@main</a>
</pre>

darinadler reviewed (approved):
I wonder if there is indeed a practical problem with old iOS apps using `showModalDialog`. Do we know of any? The safe assumption is that there are some, but I wonder if that assumption is correct.

I can’t tell where our tests to prevent us from breaking those older apps that still have `showModalDialog` are. With so many tests removed, I can’t help but think that having some around with a flag to turn the feature on would be valuable until we remove the function entirely. Removing `showModalDialog` is one way to break an old iOS app. Breaking the `showModalDialog` implementation because it’s insufficiently regression-tested is another way to make the same mistake.

rniwa reviewed (approved):
r=me

cdumez committed:
Stop exposing showModalDialog(), behind a linked-on-after check
https://bugs.webkit.org/show_bug.cgi?id=243099
<rdar://97441232>

Reviewed by Ryosuke Niwa and Darin Adler.

Now that we support the dialog element, drop support for showModalDialog(),
behind a linked-on-after check. Blink and Gecko have already dropped their
support for showModalDialog() a long time ago.

* LayoutTests/editing/execCommand/show-modal-dialog-during-execCommand.html:
* LayoutTests/fast/animation/request-animation-frame-during-modal.html:
* LayoutTests/fast/dom/Window/delete-operations.html:
* LayoutTests/fast/dom/Window/forbid-showModalDialog.html:
* LayoutTests/fast/dom/Window/open-window-min-size.html:
* LayoutTests/fast/dom/Window/showModalDialog-mandatory-parameter.html:
* LayoutTests/fast/dom/Window/window-function-frame-getter-precedence.html:
* LayoutTests/fast/dom/Window/window-function-name-getter-precedence.html:
* LayoutTests/fast/dom/Window/window-lookup-precedence.html:
* LayoutTests/fast/dom/Window/window-property-shadowing.html:
* LayoutTests/fast/dom/null-page-show-modal-dialog-crash.html:
* LayoutTests/fast/dom/wrapper-identity.html:
* LayoutTests/fast/events/beforeunload-showModalDialog.html:
* LayoutTests/fast/events/pagehide-showModalDialog.html:
* LayoutTests/fast/events/scroll-event-during-modal-dialog.html:
* LayoutTests/fast/events/show-modal-dialog-onblur-onfocus.html:
* LayoutTests/fast/events/unload-showModalDialog.html:
* LayoutTests/fast/harness/show-modal-dialog.html:
* LayoutTests/http/tests/cookies/document-cookie-after-showModalDialog.html:
* LayoutTests/http/tests/security/cross-frame-access-call.html:
* LayoutTests/http/tests/security/cross-frame-access-get.html:
* LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor.html:
* LayoutTests/http/tests/security/cross-frame-access-put.html:
* LayoutTests/http/tests/security/cross-origin-modal-dialog-base.html:
* LayoutTests/http/tests/security/navigate-when-restoring-cached-page.html:
* LayoutTests/http/tests/security/showModalDialog-sync-cross-origin-page-load.html:
* LayoutTests/http/tests/security/showModalDialog-sync-cross-origin-page-load2.html:
* LayoutTests/js/dom/function-length.html:
* Source/WTF/Scripts/Preferences/WebPreferencesInternal.yaml:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
(WebCore::JSDOMWindowBase::finishCreation):
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultShowModalDialogEnabled):
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.h:
* Source/WebKitLegacy/mac/WebView/WebPreferencesDefaultValues.mm:
(WebKit::defaultShowModalDialogEnabled):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
(-[ModalDelegate _webViewRunModal:]): Deleted.
(-[ModalDelegate _webView:setResizable:]): Deleted.
(-[ModalDelegate webView:createWebViewWithConfiguration:forNavigationAction:windowFeatures:]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestUIClient.cpp:
(beforeAll):
(testWebViewAllowModalDialogs): Deleted.
(testWebViewDisallowModalDialogs): Deleted.
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):

Canonical link: https://commits.webkit.org/252759@main
